sphinx/texinputs/sphinx.sty

```diff
@@ -706,56 +707,84 @@
 \newif\ifspx@inframed % flag set if we are already in a framed environment
 % if forced use of minipage encapsulation is needed (e.g. table cells)
 \newif\ifsphinxverbatimwithminipage   \sphinxverbatimwithminipagefalse
-\newcommand\spx@colorbox [2]{%
-% #1 will be \fcolorbox or, for first part of frame: \spx@fcolorbox
+\newcommand\spx@colorbox [3]{%
 % let the framing obey the current indentation (adapted from framed.sty's code).
   \hskip\@totalleftmargin
   \hskip-\fboxsep\hskip-\fboxrule
-  #1{VerbatimBorderColor}{VerbatimColor}{#2}%
+  \spx@fcolorbox{VerbatimBorderColor}{VerbatimColor}{#1}{#2}{#3}%
   \hskip-\fboxsep\hskip-\fboxrule
   \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth
 }
 % use of \color@b@x here is compatible with both xcolor.sty and color.sty
-\def\spx@fcolorbox #1#2%
-  {\color@b@x {\fboxsep\z@\color{#1}\spx@VerbatimFBox}{\color{#2}}}%
+\long\def\spx@fcolorbox #1#2#3#4%
+  {\color@b@x {\color{#1}\spx@VerbatimFBox{#3}{#4}}{\color{#2}}}%
 
 % Inspired and adapted from framed.sty's \CustomFBox with extra handling
 % of a non separable by pagebreak caption.
-% The sole purpose of this macro is to add the framing and the title. The #1
+% The sole purpose of this macro is to add the framing and the title. The #3
 % already has the typeset contents with background color, and took into
-% account the \sphinxverbatimsep. The \fboxsep parameter is here zero
-% (see \spx@fcolorbox definition) in order for the frame borders to exactly
-% fit the colored contents.
-\long\def\spx@VerbatimFBox#1{%
+% account the \sphinxverbatimsep.
+% #1 will hold the title or above frame material and #2 is below frame material
+\long\def\spx@VerbatimFBox#1#2#3{%
   \leavevmode
   \begingroup
-  \setbox\@tempboxa\hbox{\kern\fboxsep{#1}\kern\fboxsep}%
-   \hbox
-   {\lower\dimexpr\fboxrule+\fboxsep+\dp\@tempboxa
-    \hbox{%
-      \vbox{\ifvoid\spx@VerbatimTitleBox\else
-            % add the caption in a centered way above possibly indented frame
-            % hide its width from framed.sty's measuring step
-            % note that the caption brings \abovecaptionskip top vertical space
-            \moveright\dimexpr\fboxrule+.5\wd\@tempboxa
-                  \hb@xt@\z@{\hss\unhcopy\spx@VerbatimTitleBox\hss}\fi
+  \setbox\@tempboxa\hbox{{#3}}% contents with background color
```

Comment:
The extra brace pair: `\hbox{{#3}}` is important because `#3` contains some `\color` command for the background. Without the extra braces, this colour contaminates the border. As the background is usually white, this means the border disappears.